File: project/choisujin-onboarding/src/main/java/com/innercircle/onboarding/common/jpa/BaseEntity.java
package com.innercircle.onboarding.common.jpa;

import jakarta.persistence.*;
import lombok.Getter;
import org.hibernate.annotations.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@DynamicUpdate
@EntityListeners(AuditingEntityListener.class)
@MappedSuperclass
public abstract class BaseEntity {

    @Comment("생성일시")
    @CreatedDate
    @Column(name = "created_at", updatable = false, nullable = false, columnDefinition = "DATETIME(3)")
    private LocalDateTime createdAt;

    @Comment("수정일시")
    @LastModifiedDate
    @Column(name = "modified_at", nullable = false, columnDefinition = "DATETIME(3)")
    private LocalDateTime modifiedAt;

}

File: project/soojinhan-onboarding/src/main/kotlin/onboardingproject/project/domain/BaseEntity.kt
package onboardingproject.project.domain

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

/**
 * packageName : onboardingproject.project.domain
 * fileName    : BaseEntity
 * author      : hsj
 * date        : 2025. 6. 14.
 * description :
 */

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseEntity {
    @CreatedDate
    @Column(updatable = false)
    lateinit var createdDate: LocalDateTime

    @LastModifiedDate
    lateinit var lastModifiedDate: LocalDateTime
}
